src/ir_Mitsubishi.cpp

```diff
@@ -292,6 +340,128 @@ void IRsend::sendMitsubishiAC(unsigned char data[], uint16_t nbytes,
 }
 #endif  // SEND_MITSUBISHI_AC
 
+#if DECODE_MITSUBISHI_AC
+// Decode the supplied Mitsubishi message.
+//
+// Args:
+//   results: Ptr to the data to decode and where to store the decode result.
+//   nbits:   Nr. of data bits to expect.
+//   strict:  Flag indicating if we should perform strict matching.
+// Returns:
+//   boolean: True if it can decode it, false if it can't.
+//
+// Status: ALPHA / Under development
+//
+// Ref: https://www.analysir.com/blog/2015/01/06/reverse-engineering-mitsubishi-ac-infrared-protocol/
+bool IRrecv::decodeMitsubishiAC(decode_results *results, uint16_t nbits,
+                              bool strict) {
+  if (results->rawlen < ((MITSUBISHI_AC_STATE_LENGTH * 8 * 2) + 2)) {
+    DPRINTLN("Shorter than shortest possibly expected.");
+    return false;  // Shorter than shortest possibly expected.
+  }
+  if (strict && nbits != MITSUBISHI_AC_BITS) {
+    DPRINTLN("Request is out of spec.");
+    return false;  // Request is out of spec.
+  }
+  uint16_t offset = OFFSET_START;
+  for (uint8_t i = 0; i < MITSUBISHI_AC_STATE_LENGTH; i++) {
+    results->state[i] = 0;
+  }
+  bool failure = false;
+  uint8_t rep = 0;
+  do {
+    failure = false;
+// Header:
+//  Somtime happens that junk signals arrives before the real message
+    bool headerFound = false;
+    while (!headerFound && offset <
+        (results->rawlen - (MITSUBISHI_AC_STATE_LENGTH * 8 * 2 + 2))) {
+      headerFound = matchMark(results->rawbuf[offset++], MITSUBISHI_AC_HDR_MARK)
+          && matchSpace(results->rawbuf[offset++], MITSUBISHI_AC_HDR_SPACE);
+    }
+    if (!headerFound) {
+      DPRINTLN("Header mark not found.");
+      failure = true;
```

Comment:
Why not just `return false;` instead?
Same for almost every following `failure = false;` too.
I'd suggest failing hard early, as it saves processing time on a wasted decode.

Reply:
I use the `failure` because the repeat could help to interpret the messages - if we are not in strict mode. If I return here then I can't benefit from repeat.

Reply:
Fair enough. Though I think relying on the repeat to decode successfully is a bit redundant. We should get it the first time.